src/cp_control_types.F

```diff
@@ -201,6 +201,7 @@ MODULE cp_control_types
       REAL(KIND=dp)                        :: strength
       REAL(KIND=dp)                        :: phase_offset
       REAL(KIND=dp)                        :: wavelength
+      REAL(KIND=dp), DIMENSION(:), POINTER :: vec_pot_initial
```

Comment:
Can you turn that one into an ALLOCATABLE? Apparently, you do not really need the features of a POINTER. (Replace ASSOCIATED with ALLOCATED in the respective occasion). ALLOCATABLEs are safer in use than POINTERs.